src/rgb/contract/conceal.rs

```diff
@@ -22,21 +23,66 @@ pub trait AutoConceal {
     fn conceal_except(&mut self, seals: &Vec<seal::Confidential>) -> usize;
 }
 
-// TODO: Do an auto implementation for and other collection types
-
 impl<T> AutoConceal for Vec<T>
+where
+    T: AutoConceal,
+{
+    fn conceal_except(&mut self, seals: &Vec<seal::Confidential>) -> usize {
+        self.iter_mut()
+            .fold(0usize, |_, item| item.conceal_except(seals))
```

Comment:
This will not return a sum! It will return only the number of the concealed items in the last call. Pls see the code that I wrote in last review which is correct. Why you removed the first closure arg?

Reply:
Because as per the definition of fold here https://doc.rust-lang.org/std/iter/trait.Iterator.html I thought it will auto accumulate over the whole iteration. I didn't need to store it and the first argument was giving `unused variable` warnings so I removed it.  You are right, it does only return the last value. Changing it.